packages/x-date-pickers/src/DateCalendar/useCalendarState.tsx

```diff
@@ -54,6 +55,21 @@ export const createCalendarStateReducer =
           isMonthSwitchingAnimating: !reduceAnimations,
         };
 
+      case 'changeMonthTimezone': {
```

Comment:
I'm not convinced by the name of this action.
If it runs every time the `referenceDate` updates, could it be named `changeReferenceDate`?
Or if we want to keep the focus on the month, `changeCurrentMonthIfTimezoneChanged`?

Reply:
> If it runs every time the `referenceDate` updates, could it be named `changeReferenceDate`?

I don't have a strong preference for the action name.
It depends on how we look at the naming:
- Should it refer to what triggered the action?
- Should it refer to what the action will do?

> Or if we want to keep the focus on the month, `changeCurrentMonthIfTimezoneChanged`?

But essentially, it does not change the `currentMonth` most of the time, unless it would cause the month to change because of a timezone change. 🤔

Reply:
If you pass the timezone dirrectly to `changeMonthTimezone` instead of passing the new month, the name and its behavior would then be coherent IMHO